server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/WhiteListManager.java

```diff
@@ -752,9 +754,16 @@ private void initSqlQueries(Connection conn, org.apache.mailet.MailetContext mai
             }
 
             /*
-      Holds value of property sqlFile.
-     */
-            File sqlFile = new File((String) mailetContext.getAttribute("confDir"), "sqlResources.xml").getCanonicalFile();
+                Holds value of property sqlFile.
+            */
+            String confDir = mailetContext
+                    .getAttribute(AttributeName.of("confDir"))
```

Comment:
Can we agree that this is not working?

`MailetContext::setAttribute` is unused so this will always be empty...

Reply:
I would suggest to read this from the mailet configuration...

Reply:
I'm not sure to understand, can you show me an example please?

Reply:
I mean let's use `getInitParameter` here, like for instance it is done in `RemoveMailAttribute`

Reply:
I think it can be set in the `DataSource`, we may keep it.